core/cog-launcher.c

```diff
@@ -117,12 +117,38 @@ on_permission_request (G_GNUC_UNUSED WebKitWebView *web_view,
     return TRUE;
 }
 
+static gboolean
+on_load_failed_with_tls_errors (WebKitWebView       *web_view,
+                                char                *failing_uri,
+                                GTlsCertificate     *certificate,
+                                GTlsCertificateFlags errors,
+                                void                *user_data)
+{
+    CogShell *shell = COG_SHELL (user_data);
+    WebKitWebContext *context = webkit_web_view_get_context (web_view);
+    gboolean ignore_tls_errors = FALSE;
+    g_object_get (shell, "ignore-tls-errors", &ignore_tls_errors, NULL);
+
+    if (ignore_tls_errors) {
+        SoupURI *soup_uri = soup_uri_new (failing_uri);
```

Comment:
Please use `g_autoptr` here:

```c
g_autoptr(SoupURI) soup_uri = soup_uri_new (failing_uri);
/* ... */
```

Then you can remove the line below which calls `soup_uri_free()`.